### ISSUE TITLE
scientific notation with variables does not work

### ISSUE THREAD

**Issue by fl0wmaster** [axkr/symja_android_library#58]
# Submit a feature request or bug report

Replace any "X" with your information.

---

### Is this a request for a feature or a bug report?

Bug Report

### What is the current behavior?

6.23E-23+x
you get 'x'

### If this is a bug report, please provide steps to reproduce:

evaluate:
6.23E-23+x

### What is the expected or desired behavior?  

6.23E-23+x

### Please provide use cases for changing the current behavior:

X

### Please describe your development environment (OS, browser, etc.):

I used Android studio environment (3.0.1.), and the latest symja release (symja_java8-2018-06-18), that I embedded into my app

### Other relevant information:

X

**Comment by fl0wmaster** [comment]
Bug Report

Scientific notation with variables does not work. 

If you evaluate:
6.23E-23+x
you get 'x' instead of the expected '6.23E-23+x'
If you evaluate 6.23E-23 it does give 6.23E-23 as expected, (this was indeed fixed from previous versions where a 0 resulted).

I used Android studio environment (3.0.1.), and the latest symja release (symja_java8-2018-06-18), that I embedded into my app.

**Comment by jsveiga** [comment]
Hi fl0wmaster (not the developer, just a lib user).

It works for me:

![screenshot_20180625_120742](https://user-images.githubusercontent.com/7563634/41860016-ea94210c-7873-11e8-98dc-3acd2960b0b3.png)

TBH I normally use the java7 branch (as I want backwards compatibility to API 23 minSdk), but I just tried with minSdk 24 and the java8 latest compile, and it works the same).

Just to make sure, can you try 
(6.23E-23)+x
(as you can see in my logcat, I "encapsulate" terms in (), but because my app is an RPN calculator)

**Comment by axkr** [comment]
This seems to be related to #48.

You can set another `Config.DOUBLE_TOLERANCE` instead of `org.hipparchus.util.Precision.EPSILON`.
Otherwise Symja assumes a zero value too fast.

As mentioned in #48 the `toString` method for expressions uses a `DecimalFormat`.

See this JUnit test:
```
				Config.DOUBLE_TOLERANCE = Precision.SAFE_MIN;
				check("(6.23E-23)+x", //
						"0.0+x");
				checkNumeric("6.23E-23+x", //
						"6.23E-23+x");
```

**Comment by fl0wmaster** [comment]
His Ax Kr,

The DOUBLE_TOLERANCE seems to do the trick. I will do some more exhaustive
testing over the next few days, and let you know if it breaks.

Thanks very much for the quick response.


Sincerely,

Jan Koene
C: 408-768-0412

**Comment by fl0wmaster** [comment]
Hi jsveiga,

Thanks for your response, the bracket will not help in my case but thanks
very much for the response.

Slick looking calculator you have there, I am working on a calculator
myself, it is of topic but I am curious as to when do you plan to release?


Sincerely,

**Comment by axkr** [comment]
Here is an suggestion to mimic the Mathematica output format in the future.
See the [DoubleToMMAFormat#doubleToMMA()](https://gist.github.com/axkr/4e3e895505e75ed32a52128d1862a267) method.

May be an additional option in the future to avoid the standard Java double notation, because Symja use `E`  as a constant symbol for [Euler's constant](https://en.wikipedia.org/wiki/E_(mathematical_constant)).

**Comment by jsveiga** [comment]
Hi fl0wmaster,

I apologize; I missed your reply somehow :-(

I'll release it soon (I haven't been developing it for months, but I use it almost daily - when it gets usable, I get lazy. I still need to make the icon, then figure out my way into the Play Store or F-Droid (I've never published there, I'm old, I've published on the Ovi (Nokia) Store...). It will be 100% free and open source.

cheers,

**Comment by fl0wmaster** [comment]
Hello,

Thanks for the reply, I am looking forward to the latest version.

I use the calculator I developed, daily myself, but have not released it
either, same issue I have to look into the mechanisms in place at the
android store.

The calculator app is a fully orthogonal design where your library is fully
integrated, meaning even the 3D graphics plotting (I developed) can make
use of all the functions in your library.

I hope to release it someday in the future, but like you I am on and off on
the development depending on my time.

In any event, thanks for the update.
